### PR TITLE
Prac 05 code review request 

### PR DESCRIPTION
**Pull request by Riston-gitting-gud**
Please see if my code has any errors.
@MIKELEONGZHIFAI 
@JevanChandra
@a-rosa

### CODE REVIEW

**Comment by MikeLeongZhiFai** [comment]
yoyo test

**Review by MikeLeongZhiFai — COMMENTED** [review]
I think "email_to_name" doesnt make sense, maybe can change to something else?

**Review by MikeLeongZhiFai — COMMENTED** [review]
Code worksS

File: prac_05/emails.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+
+
+
+
+
+
+
+
+
```
Comment (JevanChandra) [inline prac_05/emails.py]: Great Code! Works well

File: prac_05/hex_colours.py [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+                   "Black": "#000000", "Carmine": "#960018", "Fulvous": "#e48400", "Goldenrod1": "ffc125",
+                   "Indigo": "4b0082", "Magenta3": "cd00cd"}
+
+colour_name = input("Enter a colour name: ").title()
+while colour_name != "":
+    print(f"The code for {colour_name} is {COLOUR_AND_CODE.get(colour_name)}")
+    colour_name = input("Enter a colour name: ").title()
+
+
+
```
Comment (JevanChandra) [inline prac_05/hex_colours.py:14]: Good code very short and simple! 👍

File: prac_05/state_names.py [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+        break
+    try:
+        state_name = CODE_TO_NAME[state_code]
+    except KeyError:
+        print("Invalid short state")
+    else:
+        print(f"{state_code} is {state_name}")
+
+for state_code, state_name in CODE_TO_NAME.items():
+    print(f"{state_code:<3} is {state_name}")
```
Comment (JevanChandra) [inline prac_05/state_names.py:25]: Nice code. Did not think of using try.

File: prac_05/word_occurrences.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    word_to_count[word] = word_to_count.get(word, 0) + 1
+
+words = list(word_to_count.keys())
+words.sort()
+
+max_length = max(len(word) for word in words)
+for word in words:
+    print(f"{word:{max_length}} : {word_to_count[word]}")
+
+
```
Comment (JevanChandra) [inline prac_05/word_occurrences.py:20]: Works well!

File: prac_05/wimbledon.py [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    records = []
+    with open(filename, "r", encoding="utf-8-sig") as in_file:
+        in_file.readline()
+        for line in in_file:
+            parts = line.strip().split(",")
+            records.append(parts)
+    return records
+
+
+main()
```
Comment (JevanChandra) [inline prac_05/wimbledon.py:48]: Works very well!

File: prac_05/wimbledon.py [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    display_results(champion_to_count, countries)
+
+
+def process_records(records):
+    """Create dictionary of champions and set of countries from records (list of lists)."""
+    champion_to_count = {}
+    countries = set()
+    for record in records:
+        countries.add(record[COUNTRY_INDEX])
+        try:
```
Comment (a-rosa) [inline prac_05/wimbledon.py:21]: You could use if else instead of try except

File: prac_05/wimbledon.py [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+            champion_to_count[record[CHAMPION_INDEX]] += 1
+        except KeyError:
+            champion_to_count[record[CHAMPION_INDEX]] = 1
```
Comment (a-rosa) [inline prac_05/wimbledon.py:24]: You could use if else instead of try except

File: prac_05/wimbledon.py [hunk 8]
```diff
@@ -0,0 +1,48 @@
+# Game, Set, Match
+
```
Comment (a-rosa) [inline prac_05/wimbledon.py:2]: If I am not wrong, you also need to add estimate time and actual time here

File: prac_05/word_occurrences.py [hunk 9]
```diff
@@ -1 +1,20 @@
...
```
Comment (a-rosa) [inline prac_05/word_occurrences.py:20]: You could add some comments on some lines to make readers understand what it does

File: prac_05/emails.py [hunk 10]
```diff
@@ -1,3 +1,37 @@
...
+    parts = prefix.split(".")
+    name = ""
+    for part in parts:
+        name += part.capitalize()
+    return name
+
+
+main()
+
+
```
Comment (a-rosa) [inline prac_05/emails.py]: Too much empty lines

File: prac_05/emails.py [hunk 11]
```diff
@@ -1,3 +1,37 @@
...
+            name = input("Name: ")
+        email_to_name[email] = name
+        email = input("Email: ")
+    for email, name in email_to_name.items():
+        print(f"{name} {email}")
+
+
+def get_name_from_email(email):
+    prefix = email.split("@")[0]
+    parts = prefix.split(".")
```
Comment (a-rosa) [inline prac_05/emails.py:20]: You could mix both prefix and parts together:
emil.split("@")[0].split(".")

File: prac_05/hex_colours.py [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+COLOUR_AND_CODE = {"Aliceblue": "#0048ba", "Antiquewhite": "#faebd7", "Aqua": "#00ffff", "Bistre": "#3d2b1f",
+                   "Black": "#000000", "Carmine": "#960018", "Fulvous": "#e48400", "Goldenrod1": "ffc125",
+                   "Indigo": "4b0082", "Magenta3": "cd00cd"}
+
+colour_name = input("Enter a colour name: ").title()
+while colour_name != "":
+    print(f"The code for {colour_name} is {COLOUR_AND_CODE.get(colour_name)}")
+    colour_name = input("Enter a colour name: ").title()
+
```
Comment (a-rosa) [inline prac_05/hex_colours.py:12]: There would be an error if the user don't enter a color that is in the set

File: prac_05/state_names.py [hunk 13]
```diff
@@ -0,0 +1,25 @@
...
+State names in a dictionary
+File has been reformatted
+"""
+
+# TODO: Reformat this file so the dictionary code follows PEP 8 convention
+CODE_TO_NAME = {"QLD": "Queensland", "NSW": "New South Wales", "NT": "Northern Territory", "WA": "Western Australia",
+                "ACT": "Australian Capital Territory", "VIC": "Victoria", "TAS": "Tasmania"}
+print(CODE_TO_NAME)
+
+
```
Comment (a-rosa) [inline prac_05/state_names.py:12]: You could just use while state_code != "" and before that ask for an input so there is no need to have a break in it

**Pull request merged**